.github/workflows/ci.yaml

```diff
@@ -0,0 +1,93 @@
+---
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - README.md
+      - .gitignore
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository_owner }}/comment
+
+jobs:
+  version:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Get new version
+        id: version_tag
+        uses: anothrNick/github-tag-action@8c8163ef62cf9c4677c8e800f36270af27930f42 # 1.61.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          TAG_CONTEXT: branch
+          PRERELEASE: ${{ github.ref != 'refs/heads/main' }}
+  
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.x
+      - name: Install dependencies
+        run: npm install
+      - name: Run tests
+        run: npm test
+
+  build:
+    needs:
+      - version
+      - test
+    permissions:
+      packages: write
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Extract metadata (tags, labels) for Docker
+      id: meta
+      uses: docker/metadata-action@v4
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+    - uses: actions-ecosystem/action-get-latest-tag@b7c32daec3395a9616f88548363a42652b22d435 # v1.6.0
+      id: latest_tag
+
+    - name: Build and push image tag ${{steps.latest_tag.outputs.tag}}
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        platforms: linux/amd64,linux/arm64
+        push: true
+        labels: ${{ steps.meta.outputs.labels }}
+        tags: |
+          ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{steps.latest_tag.outputs.tag}}
```

Comment:
If primary reason is to have the versioning at this step, we could use commit id for non-protected branches. 
That is if the image should be pushed at all - I tend to do commits and think afterwards, especially if working on something mid-to-large-sized.

Reply:
ah, sorry, you do mean the commit sha :)

Reply:
>I tend to do commits and think afterwards
Well, pushing beta builds to the registry doesn't need to think more tbh, it doesn't make me lol

Reply:
That being said, it might not be the best practice, true. But also the unstable builds are tagged with `-beta` in the name, so it's kind of clear what is unstable af. We could see what we want there though.